modules/operator/main.tf

```diff
@@ -0,0 +1,188 @@
+      var.environment,
+      each.key,
+      coalesce(each.value.namespace, kubernetes_namespace.materialize.metadata[0].name),
+      each.value.instance_id
+    )
+  }
+}
+
+resource "kubernetes_manifest" "materialize_instances" {
+  for_each = { for idx, instance in var.instances : instance.name => instance }
+
+  manifest = {
+    apiVersion = "materialize.cloud/v1alpha1"
+    kind       = "Materialize"
+    metadata = {
+      name      = each.value.instance_id
+      namespace = coalesce(each.value.namespace, var.operator_namespace)
+    }
+    spec = {
+      environmentdImageRef = "materialize/environmentd:${var.operator_version}"
+      backendSecretName    = "${each.key}-materialize-backend"
+      environmentdResourceRequirements = {
+        limits = {
+          memory = each.value.memory_limit
+        }
+        requests = {
+          cpu    = each.value.cpu_request
+          memory = each.value.memory_request
+        }
+      }
+      balancerdResourceRequirements = {
+        limits = {
+          memory = "256Mi"
+        }
+        requests = {
+          cpu    = "100m"
+          memory = "256Mi"
+        }
+      }
+    }
+  }
+
+  depends_on = [
+    helm_release.materialize_operator,
+    kubernetes_secret.materialize_backends,
+    kubernetes_namespace.instance_namespaces
+  ]
+}
+
+# Materialize does not currently create databases within the instances, so we need to create them ourselves
+resource "kubernetes_manifest" "db_init_configmap" {
+  for_each = { for idx, instance in var.instances : instance.name => instance }
+
+  manifest = {
+    apiVersion = "v1"
+    kind       = "ConfigMap"
+    metadata = {
+      name      = "init-db-script-${each.key}"
+      namespace = coalesce(each.value.namespace, var.operator_namespace)
+    }
+    data = {
+      "init.sql" = format(
+        "CREATE DATABASE IF NOT EXISTS %s;",
+        coalesce(each.value.database_name, "${each.key}_db")
+      )
+    }
+  }
+}
+
+resource "kubernetes_manifest" "db_init_job" {
+  for_each = { for idx, instance in var.instances : instance.name => instance }
+
+  manifest = {
+    apiVersion = "batch/v1"
+    kind       = "Job"
+    metadata = {
+      name      = "create-db-${each.key}"
+      namespace = coalesce(each.value.namespace, var.operator_namespace)
+    }
+    spec = {
+      template = {
+        spec = {
+          containers = [{
+            name  = "init-db"
+            image = "postgres:latest"
+            command = [
+              "/bin/sh",
+              "-c",
+              format(
+                "psql $DATABASE_URL -c \"CREATE DATABASE %s;\"",
+                coalesce(each.value.database_name, "${each.key}_db")
+              )
+            ]
+            env = [
+              {
+                name = "DATABASE_URL"
+                value = format(
+                  "postgres://%s:%s@%s/%s?sslmode=require",
+                  each.value.database_username,
+                  each.value.database_password,
```

Comment:
this means the pw will appear in plaintext on the Job's definition. you could inject the database username / pw / host as env vars from the Secret (which should be in the same namespace) and construct the URL in the `psql` statement